reflex/components/sonner/toast.py

```diff
@@ -0,0 +1,224 @@
+"""Sonner toast component."""
+
+from __future__ import annotations
+
+from typing import Dict, Literal, Optional
+
+from reflex.base import Base
+from reflex.components.component import Component, ComponentNamespace
+from reflex.components.lucide.icon import Icon
+from reflex.event import EventSpec, call_script
+from reflex.style import color_mode
+from reflex.utils import format
+from reflex.utils.imports import ImportVar
+from reflex.utils.serializers import serialize
+from reflex.vars import Var, VarData
+
+LiteralPosition = Literal[
+    "top-left",
+    "top-center",
+    "top-right",
+    "bottom-left",
+    "bottom-center",
+    "bottom-right",
+]
+
+
+toast_ref = Var.create_safe("refs['__toast']")
+
+
+class PropsBase(Base):
+    """Base class for all props classes."""
+
+    def json(self) -> str:
+        """Convert the object to a json string.
+
+        Returns:
+            The object as a json string.
+        """
+        from reflex.utils.serializers import serialize
+
+        return self.__config__.json_dumps(
+            {format.to_camel_case(key): value for key, value in self.dict().items()},
+            default=serialize,
+        )
+
+
+class ToastProps(PropsBase):
+    """Props for the toast component."""
+
+    description: str = ""
+    close_button: bool = False
+    invert: bool = False
+    important: bool = False
+    duration: int = 4000
+    position: LiteralPosition = "bottom-right"
+    dismissible: bool = True
+    icon: Optional[Icon] = None
+    action: str = ""
+    cancel: str = ""
```

Comment:
it doesn't seem like the action/cancel actually works, maybe we should remove these for now until we can figure out a way to chain events from them.